src/api/server.rkt

```diff
@@ -252,26 +275,73 @@
                          *loose-plugins*)
    (parameterize ([current-error-port (open-output-nowhere)]) ; hide output
      (load-herbie-plugins))
+   (define worker-thread
+     (thread (λ ()
+               (let loop ([seed #f])
+                 (match (thread-receive)
+                   [`(init rand
+                           ,vec
+                           flags
+                           ,flag-table
+                           num-iters
+                           ,iterations
+                           points
+                           ,points
+                           timeout
+                           ,timeout
+                           output-dir
+                           ,output
+                           reeval
+                           ,reeval
+                           demo?
+                           ,demo?)
+                    (set! seed vec)
+                    (*flags* flag-table)
+                    (*num-iterations* iterations)
+                    (*num-points* points)
+                    (*timeout* timeout)
+                    (*demo-output* output)
+                    (*reeval-pts* reeval)
+                    (*demo?* demo?)]
+                   [job-info (run-job job-info)])
+                 (loop seed)))))
```

Comment:
Where does the thread send the results back?

Reply:
Ok, so I wasn't sure if there was a better way to do this. 
But right now I am passing in a reference to the `manager` through `job-info`. Which allows the result to be reported back with `(place-channel-put manager (list 'finished manager worker-id job-id out-result))` in `run-job`. Line 344.